docs: clarify consistent-type-imports guidance for verbatimModuleSyntax (#12194)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/consistent-type-imports.mdx

## Comparison with `importsNotUsedAsValues` / `verbatimModuleSyntax`

[`verbatimModuleSyntax`](https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax) was introduced in TypeScript v5.0 (as a replacement for `importsNotUsedAsValues`).
This rule and `verbatimModuleSyntax` overlap in some situations, but there are a few behavior differences:
| Situation | `consistent-type-imports` (ESLint) | `verbatimModuleSyntax` (TypeScript) |
| -------------------------------------------------------------- | --------------------------------------------------------- | ----------------------------------------------------------- |
| Unused imports | Ignored (consider using [`@typescript-eslint/no-unused-vars`](/rules/no-unused-vars)) | Type error |
| Usage with `emitDecoratorMetadata` & `experimentalDecorators` | Ignores files that contain decorators | Reports on files that contain decorators |
| Failures detected | Does not fail `tsc` build; can be auto-fixed with `--fix` | Fails `tsc` build; cannot be auto-fixed on the command-line |
| `import { type T } from 'T';` | TypeScript will emit nothing (it "elides" the import) | TypeScript emits `import {} from 'T'` |

> Because there are some differences, using both this rule and `verbatimModuleSyntax` at the same time can lead to duplicate reports on the same import.

Reviewer: The new version is a non sequitur as written: the differences between them don't cause _duplicate_ reports (it's the similarities that do). The original is more accurate.

perhaps we can succinctly incorporate both like so?

```suggestion
Because they have slightly different goals, using both this rule and `verbatimModuleSyntax` at the same time can lead to conflicting and/or duplicate reports on the same import.
```
